### ISSUE TITLE
Implement sending a Document via XMLHttpRequest

### ISSUE THREAD

**Issue by jdm** [servo/servo#9963]
Specification: https://xhr.spec.whatwg.org/#interface-xmlhttprequest and https://xhr.spec.whatwg.org/#dom-xmlhttprequest-send
Code: `components/script/dom/webidls/XMLHttpRequest.webidl`, `components/script/dom/xmlhttprequest.rs` (see `Send` method)
Tests: `./mach test-wpt tests/wpt/web-platform-tests/XMLHttpRequest/`
Minimal testcase:
```html
<script>
var x = new XMLHttpRequest();
x.open('POST', location.href, false);
x.send(document);
</script>
```
And run that from an http server like [this minimal python one](https://gist.github.com/jdm/d4215dd7eee6650cec58), where you can inspect the data received in the terminal window.

For implementing the serialization of the document, we should be able to use the existing HTML serialization code. See `serialize` in `components/script/dom/element.rs` for an example!

**Comment by jdm** [comment]
Note: this is reserved for a potential RGSoC applicant, so please talk with me before starting to work on this!

**Comment by jdm** [comment]
Closing in favour of #9490.
